components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/authz/OAuth2AuthzEndpoint.java

```diff
@@ -252,6 +249,12 @@ public Response authorize(@Context HttpServletRequest request, @Context HttpServ
             throw e;
         }
 
+        // Start tenant domain flow if the tenant configuration is not enabled.
+        if (!IdentityTenantUtil.isTenantedSessionsEnabled()) {
+            String tenantDomain = EndpointUtil.getSPTenantDomainFromClientId(oAuthMessage.getClientId());
+            FrameworkUtils.startTenantFlow(tenantDomain);
+        }
```

Comment:
let's move these lines inside the below try block.

Reply:
Fixed by b65ad95a158ea477d1c6b3d4362b836ff4f88dd1.